code/game/machinery/ftl_shieldgen.dm

```diff
@@ -189,20 +204,85 @@
 		raise_physical()
 	else if(!is_active() && shield_barrier_objs.len)
 		drop_physical()
+		plasma_charge = 0
+
 
 /obj/effect/ftl_shield
 	name = "Shield"
 	desc = "A powerful ship shields. A simple hand weapon would not be enough to take it out."
-	icon = 'icons/effects/effects.dmi'
-	icon_state = "shieldwall"
+	icon = 'icons/obj/machines/shielding.dmi'
+	icon_state = "shield"
+	layer = ABOVE_MOB_LAYER
 	density = 1
+	alpha = 230
 	var/in_dir = 2
 	anchored = 1
 	pass_flags = LETPASSTHROW
 
+	var/list/adjacent_shields_dir = list()
+
 /obj/effect/ftl_shield/CanPass(atom/movable/mover, turf/target, height=0) // Shields are one-way: Shit can leave, but shit can't enter
 	if(istype(loc, /turf/open/space/transit))
 		return 0
 	if(get_dir(src, target) == in_dir)
 		return 1
 	return 0
+
+/obj/effect/ftl_shield/Initialize()
+	set_adjacencies(TRUE)
+	..
+	update_icon()
+
+obj/effect/ftl_shield/Destroy()
+	set_adjacencies(TRUE)
+	..
+
+/obj/effect/ftl_shield/attack_hand(var/mob/living/user)
+	impact_effect(3) // Harmless, but still produces the 'impact' effect.
+	..()
+
+/obj/effect/ftl_shield/CollidedWith(atom/A)
+	..(A)
+	impact_effect(2)
+
+
+/obj/effect/ftl_shield/update_icon(var/update_neighbors = 0)
+	overlays.Cut()
+	set_adjacencies(update_neighbors)
+	if(density)
+		icon_state = "shield"
+		set_light(3, 3, "#66FFFF")
+	else
+		icon_state = "shield_broken"
+		set_light(3, 5, "#FF9900")
+
+	// Edge overlays
+	for(var/found_dir in adjacent_shields_dir)
+		add_overlay(image(src.icon, src, "shield_edge", dir = found_dir))
+
+
+// Small visual effect, makes the shield tiles 	brighten up by becoming more opaque for a moment, and spreads to nearby shields.
+obj/effect/ftl_shield/proc/impact_effect(var/i, var/list/affected_shields = list())
+	alpha = 175
+	animate(src, alpha = initial(alpha), time = 1)
+	affected_shields |= src
+	i--
+	if(i)
+		addtimer(CALLBACK(src, .proc/spread_impact, affected_shields), 2)
+
+obj/effect/ftl_shield/proc/set_adjacencies(var/update_neighbors)
+	for(var/direction in GLOB.cardinals)
+		var/turf/T = get_step(loc, direction)
+		if(T) // Incase we somehow stepped off the map.
+			for(var/obj/effect/ftl_shield/F in T)
+				if(update_neighbors)
+					F.update_icon()
+				adjacent_shields_dir |= direction
+
+obj/effect/ftl_shield/proc/spread_impact(var/i, var/list/affected_shields = list())
+	for(var/direction in GLOB.cardinals)
+		var/turf/T = get_step(src, direction)
+		if(T) // Incase we somehow stepped off the map.
+			for(var/obj/effect/ftl_shield/F in T)
+				if(!(F in affected_shields))
```

Comment:
affected_shields should be an associative list so you can check if a shield is in it with affected_shields[F]